Выполнены задания 1-6 + Задачи на Generic

Roman8x opened the pull request.

Roman8x committed:
Выполнены задания 1-6

Valeran86 commented on src/main/java/ru/sbt/collections/Counter1DifferentWords.java:
```diff
@@ -17,5 +24,110 @@ public static void main( String[] args ) throws IOException, URISyntaxException
+
+   public static void task1 (String text){
+        if (text == null) return ;
+       Set  <String> set = new HashSet(Arrays.asList(text2words (text))) ;
```
обычное правило всех команд разработки - 1-2 действия одна строка, т.е. удобнее читать вот так:
```
if ( text == null ) 
  return;
String[] words = text2words( text );
Set<String> set = new HashSet<>( Arrays.asList( words  ) ) ;
```

Valeran86 commented on src/main/java/ru/sbt/collections/Counter1DifferentWords.java:
```diff
@@ -17,5 +24,110 @@ public static void main( String[] args ) throws IOException, URISyntaxException
+                    @Override
+                    public int compare(String o1, String o2) {
+                     int res = o1.length() - o2.length();
+                     if (res==0) return 0;
```
вот тут кажется ошибка, в итоге в этом задании должно получаться ~490 слов.

Valeran86 commented on src/main/java/ru/sbt/collections/Counter1DifferentWords.java:
```diff
@@ -17,5 +24,110 @@ public static void main( String[] args ) throws IOException, URISyntaxException
+        HashMap <String, Integer> hs_word_count = new HashMap<>();
+
+        for (String word:  text2words (text)) {
+            hs_word_count.merge(word, 1,Integer::sum);
```
отличное решение!

Roman8x committed:
Выполнены задания 1-6
1. Добавлена проверка строк одинаковой длины через compareTo
2. Код разнесен на несколько  строчек
3. Задача 4. Убрал преборазование к списку

Roman8x commented:
Замечания учтены

Roman8x committed:
Задача на Generic:
Реализовать CountMap.

Roman8x committed:
Задача 2 на Generic:
Список
